app/drivers/zephyr/dts/bindings/gpio/zmk,gpio-165.yaml

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) 2022 The ZMK Contributors
+#
+# SPDX-License-Identifier: MIT
+#
+
+description: >
+  This is a representation of the 165 Parrallel-in Shift Register.
+
+compatible: "zmk,gpio-165"
+
+include: [gpio-controller.yaml, spi-device.yaml]
+
+properties:
+  "#gpio-cells":
+    const: 2
+
+  ngpios:
+    type: int
+    required: true
+    enum:
```

Comment:
This should not be an enum, I think. 165s can be daisy-chained more than 4.

![image](https://user-images.githubusercontent.com/288288/219263459-2d12f616-941a-4b9d-9489-830bc0e37a8e.png)

Reply:
You are correct that more than 4 can be daisy-chained, however, zephyr currently only supports up to 32-bit ports, see here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/gpio.h#L525
and here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/gpio.h#L229

I'm not sure how to implement support for more than 4, but I'm think it would have to be in a way that one gpi-165 instance references another, so you can actually refer to the index>32 gpios from your device tree. Or perhaps there could be a common node that doesn't expose a gpio driver, and sub-nodes that each expose up to 32 bits per gpio port. In any case this would be more complicated than just updating the ngpios data type.

Also this PR follows almost 1:1 what the existing gpio_595 driver does, which is also limited to 32 gpios, probably for the same reason.

I think supporting more than 32 gpios is out of scope for this PR, and eventually when support for that is added, it should be added symmetrically for both 165 and 595.